modules/apps/login/login-authentication-openid-connect-web/src/main/java/com/liferay/login/authentication/openid/connect/web/internal/portlet/action/OpenIdConnectLoginRequestMVCActionCommand.java

```diff
@@ -106,6 +108,20 @@ public void doProcessAction(
 		throws Exception {
 
 		try {
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)actionRequest.getAttribute(WebKeys.THEME_DISPLAY);
+
+			if (themeDisplay.isSignedIn()) {
+				String redirect = ParamUtil.getString(
+					actionRequest, "redirect");
```

Comment:
@istvansajtos we need to escape the redirect value before using it. Can you add a commit?